kaiax/gasless/impl/init.go

```diff
@@ -45,20 +49,24 @@ func NewGaslessModule() *GaslessModule {
 	return &GaslessModule{}
 }
 
-func (g *GaslessModule) Init(opts *InitOpts) error {
-	if opts == nil || opts.ChainConfig == nil || opts.NodeKey == nil || opts.TxPool == nil {
-		return ErrInitUnexpectedNil
+func (g *GaslessModule) Init(opts *InitOpts) (disabled bool, err error) {
+	if opts == nil || opts.ChainConfig == nil || opts.CNConfig == nil || opts.NodeKey == nil || opts.Chain == nil || opts.TxPool == nil {
+		return true, ErrInitUnexpectedNil
 	}
-	g.InitOpts = *opts
 
-	g.swapRouters = map[common.Address]bool{
-		common.HexToAddress("0x1234"): true,
-	}
-	g.allowedTokens = map[common.Address]bool{
-		common.HexToAddress("0xabcd"): true,
+	if opts.CNConfig.Disable {
+		return true, nil
```

Comment:
I don't think returning `disabled` is a good idea. Other modules all have the same `Init` format.
Instead, why not introduce an interface and let GaslessModule implement it?

```
interface Disableable {
	IsDisabled() bool
}
```

Then, in `node/cn/backend.go`, we can skip `Init()` of disabled.

Reply:
Or we can simply check `--gasless-disable` flag from `node/cn/backend`

Reply:
Ok, I prefer IsDisabled(). I think that it can be organized as Initializer interface in the future.

```golang
interface InitializerModule {
    Init(interface{}) error
    IsDisabled() bool
}
```

Reply:
fixed at https://github.com/kaiachain/kaia/pull/286/commits/77cfd7d62a3fb8a11587138f8bab387e93fc6537